### PR TITLE
Availability of resources in cooperation

### LINKED ISSUE #2503: (SP:2) Fix changing availability of resources in Cooperation according to changes on backend
### Fix changing availability of resources in Cooperation according to changes on backend

Currently, in **Cooperation**, there is a mechanism of changing the `availability` of a resource in the inner field `resource`. However, after the changes that were required and done in  [PR #904](https://github.com/ita-social-projects/SpaceToStudy-BackEnd/pull/904) and [PR #907](https://github.com/ita-social-projects/SpaceToStudy-BackEnd/pull/907), we need to set/change the `availability` at the top level alongside the `resource` and `resourceType` fields for each resource included in the cooperation:

Screenshot 2024-09-11 at 23 35 47

### PR DESCRIPTION
**Pull request by OliaHrytsak**
Now the Availability of resources is moved from inside of resources to the same level as resource on UI side
 And tutor can change and save Availability of particular resource in cooperation Activities

https://github.com/user-attachments/assets/fac1f136-a050-47bf-b6a6-00e9e7126a44

image

### CODE REVIEW

File: src/containers/course-section/resources-list/ResourcesList.tsx [hunk 1]
```diff
@@ -8,12 +8,18 @@ import SortableWrapper from '~/containers/sortable-wrapper/SortableWrapper'
...
+  ResourceAvailabilityStatusEnum,
+  Resource
+} from '~/types'
 
 import useDroppable from '~/hooks/use-droppable'
 import useDndSensor from '~/hooks/use-dnd-sensor'
 
 interface ResourcesListProps {
+  cooperationData?: Resource[]
   items: CourseResource[]
```
Comment (YaroslavChuiko) [inline src/containers/course-section/resources-list/ResourcesList.tsx]: Why do we need to pass `items` together with `cooperationData` as props, as I understand from the code we can always get `items` from `cooperationData`

Comment (OliaHrytsak) [reply src/containers/course-section/resources-list/ResourcesList.tsx]: Fixed! Thank you for this suggestion

File: src/components/cooperation-section-view/CooperationSectionView.tsx [hunk 2]
```diff
@@ -20,8 +24,9 @@ const CooperationSectionView: FC<CooperationSectionViewProps> = ({ item }) => {
 
   const resources = useMemo<undefined | ReactNode[]>(
     () =>
-      item.resources?.map(({ resource, resourceType }) => (
+      item.resources?.map(({ availability, resource, resourceType }) => (
         <ResourceItem
+          availability={availability as ResourceAvailability}
```
Comment (PavloDolia) [inline src/components/cooperation-section-view/CooperationSectionView.tsx]: can we change `availability` in the `Resource` interface in the file `src/types/course/interfaces/course.interface.ts` to remove this `as ResourceAvailability`?
E. g.:
Screenshot 2024-09-17 at 12 21 48

Comment (OliaHrytsak) [reply src/components/cooperation-section-view/CooperationSectionView.tsx]: Fixed

**Review by PavloDolia — APPROVED** [review]

**Review by YaroslavChuiko — APPROVED** [review]

**Review by BohdanMylyi — APPROVED** [review]

**Pull request merged**